core/src/main/java/io/grpc/LoadBalancer2.java

```diff
@@ -317,6 +322,77 @@ public String toString() {
   }
 
   /**
+   * Provides arguments for a {@link SubchannelPicker#pickSubchannel(PickSubchannelArgs)}.
+   */
+  public abstract static class PickSubchannelArgs {
```

Comment:
That puts a bit of a burden on implementors when writing tests. Not a strong opinion but I slightly favour builder though I understand your point of view and I'm fine reverting that back. Let me know what do you think works better for you.